pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -381,6 +383,27 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation, operationType
 		return gardencorev1beta1helper.NewWrappedLastErrors(gardencorev1beta1helper.FormatLastErrDescription(err), flow.Errors(err))
 	}
 
+	// Remove completed tasks from Shoot if they were executed.
+	newShoot, err := kutil.TryUpdateShootAnnotations(c.k8sGardenClient.GardenCore(), retry.DefaultRetry, o.Shoot.Info.ObjectMeta,
+		func(shoot *gardencorev1beta1.Shoot) (*gardencorev1beta1.Shoot, error) {
+			var tasksToRemove []string
+
+			if requestInfrastructureReconciliation {
+				tasksToRemove = append(tasksToRemove, common.ShootTaskDeployInfrastructure)
+			}
+			if requestControlPlanePodsRestart {
+				tasksToRemove = append(tasksToRemove, common.ShootTaskRestartControlPlanePods)
+			}
```

Comment:
What do you think about saving all tasks before running the flow and removing them instead of doing this only for these specific ones?

Reply:
+1, will do

Reply:
done, PTAL